src/logger.c

```diff
@@ -90,17 +96,31 @@ static struct log_file pot = {NULL, NULL, NULL, 0, -1};
 
 static int in_logger = 0;
 
-static void log_file_init(struct log_file *f, char *name, int size)
+static void log_file_init(struct log_file *f, char *name, char *perms, int size)
 {
+	p_u = "0600";
+	p_ug = "0660";
+	p_ugo = "0666";
+	if (!((strcmp(perms, p_u) == 0) || (strcmp(perms, p_ug) == 0) || (strcmp(perms, p_ugo) == 0)))
+		pexit("PotFilePerms or LogFilePerms %s invalid, valid values: %s, %s, %s\n\n", perms, p_u, p_ug, p_ugo);
```

Comment:
Please note that `pexit()` should only be used for things that set errno, eg. `chmod()`, but this is not the case here. You should change it to
```c
	if (<insert Jim's clever tests here>) {
		fprintf(stderr, "PotFilePerms or LogFilePerms %s invalid\n", perms);
		error();
	}
```